internal/pkg/merge/merger.go

```diff
@@ -0,0 +1,225 @@
+package merge
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/protobom/protobom/pkg/sbom"
+)
+
+type (
+	Merger[T any] interface {
+		Merge(*T) (*T,error)
+	}
+
+	SliceMerger[T any] interface {
+		Merge([]*T) ([]*T,error)
+	}
+
+	MergerBase[T any] struct {
+		Base *T
+	}
+
+	SliceMergerBase[T any] struct {
+		Base []*T
+	}
+)
```

Comment:
I think instead of `any` it should be constrained a _little_.

```suggestion
type (
	Merger[T proto.Message] interface {
		Merge(T) (T, error)
	}

	SliceMerger[T proto.Message, S []T] interface {
		Merge(S) (S, error)
	}

	MergerBase[T proto.Message] struct {
		Base T
	}

	SliceMergerBase[T proto.Message] struct {
		Base []T
	}
)
```